workers/unity/Assets/Fps/Scripts/SetupLogic/ConnectionController.cs

```diff
@@ -12,93 +13,87 @@ public class ConnectionController : MonoBehaviour
         [Require] private PlayerCreator.Requirable.CommandRequestSender commandSender;
         [Require] private PlayerCreator.Requirable.CommandResponseHandler responseHandler;
 
-        private GameObject canvasCameraObj;
-        private ScreenUIController screenUIController;
-        private Animator connectButton;
         private WorkerConnector clientWorkerConnector;
 
         private void Start()
         {
+            ConnectionStateReporter.InformOfConnectionController(this);
             clientWorkerConnector = gameObject.GetComponent<WorkerConnector>();
-            connectButton = screenUIController.ConnectScreen.GetComponentInChildren<Animator>();
+            clientWorkerConnector.OnWorkerCreationFinished += OnWorkerCreated;
         }
 
-        private void OnEnable()
+        private void OnWorkerCreated(Worker worker)
         {
-            if (responseHandler != null)
+            if (worker.Connection.GetConnectionStatusCode() == ConnectionStatusCode.Success)
             {
-                responseHandler.OnCreatePlayerResponse += OnCreatePlayerResponse;
+                StartCoroutine(nameof(DelayedConnectedMessage));
+            }
+            else
+            {
+                ConnectionStateReporter.SetState(ConnectionStateReporter.State.ConnectionFailed,
+                    worker.Connection.GetConnectionStatusCode().ToString());
```

Comment:
Do we want to get the status code detail instead (which is a human friendly message)?

Reply:
Sounds likely.